pandas/io/parsers/python_parser.py

```diff
@@ -506,13 +510,13 @@ def _infer_columns(self):
                             "header fields in the file"
                         )
                     # Ignore output but set used columns.
-                    self._handle_usecols([names], names)
+                    self._handle_usecols([names], names, ncols)
                     columns = [names]
                     num_original_columns = ncols
 
         return columns, num_original_columns, unnamed_cols
 
-    def _handle_usecols(self, columns, usecols_key):
+    def _handle_usecols(self, columns, usecols_key, num_original_columns):
```

Comment:
Brief docstring on this new parameter to explain how it differs from `columns` (and why we couldn't just use `columns.length` in the logic).

Reply:
can you type args here